Initial submission for new VOD List Resource Type

marktrayer opened the pull request.

marktrayer committed:
Initial submission for new VOD List Resource Type

marktrayer committed:
Update oic.r.vodlist.swagger.json

mkettrick commented on swagger2.0/oic.r.vodlist.swagger.json:
```diff
@@ -0,0 +1,135 @@
+      "properties": {
+        "n": {
+            "$ref": "https://openconnectivityfoundation.github.io/core/schemas/oic.common.properties.core-schema.json#/definitions/n"
+        },
```
Mark "n" as read only?
EDIT: I see that it's marked as read only down below.  But both "di" and "econame" are also listed as read only in this section so I guess "n" needs to be as well???
EDIT 2: The "n" marked as read only below is the Common Property of the Resource, not the "n" Property of a VOD.  So I do think we must mark this as read only here as well.

marktrayer replied:
That's fine; that's the intent, I'll update.

marktrayer replied:
Fixed in latest version, please check.

mkettrick replied:
Looks good.

mkettrick commented on swagger2.0/oic.r.vodlist.swagger.json:
```diff
@@ -0,0 +1,135 @@
+        }
+      },
+      "required": ["n", "di", "econame"]
+    },
```
If "n" is blank for the VOD itself, is the intent that this Resource would expose "n": "" (i.e. a blank Property Value)?

marktrayer replied:
Yes.  'n' is mandatory for /oic/d so it has to be there.

mkettrick replied:
So we'll need spec text that ties "n" of /oic/d to "n" of this Resource.

marktrayer replied:
Agreed. Added details to the description also.  Please check.

mkettrick replied:
Looks good.

mkettrick commented on swagger2.0/oic.r.vodlist.swagger.json:
```diff
@@ -0,0 +1,135 @@
+        },
+        "di" : {
+          "description": "Device Id of VOD",
+          "allOf": [
```
Why does "di" have an "allOf" but "n" and "econame" do not?  I'm sure there is a reason but I'm not clear when it's needed and when it's not needed.  Was this a CTT thing?

marktrayer replied:
Blame Comarch.  It's because if there is a $ref all other keywords are ignored, it's a wash if just 'description' but a problem if keywords that are 'active' (i.e. readOnly).

mkettrick commented on swagger2.0/oic.r.vodlist.swagger.json:
```diff
@@ -0,0 +1,135 @@
+            ],
+            "type": "string"
+          },
+          "minItems": 1,
```
Should "minItems" be 2 in this case?  Or is it allowed to only expose one of the OCF Interfaces?

marktrayer replied:
Probably, I'll update.

mkettrick commented on swagger2.0/oic.r.vodlist.swagger.json:
```diff
@@ -0,0 +1,135 @@
+          "items": {
+            "$ref": "#/definitions/vodentry"
+          }
+        }
```
Do we also need a "required" tag here?

marktrayer replied:
Yes

marktrayer replied:
Added in latest version, please check.

mkettrick replied:
Is it the case that only "vods" is marked as required since it is exposed by both oic.if.r and oic.if.baseline?  The other Core Properties may not be exposed by oic.if.r.

marktrayer replied:
It's the only one required in all instances, correct.

joochlee replied:
what is the purpose of vodlist.id ?

mkettrick commented on swagger2.0/oic.r.vodlist.swagger.json:
```diff
@@ -0,0 +1,135 @@
+        "responses": {
+            "200": {
+              "description" : "",
+              "x-example":
```
What are "description" and "x-example"?  Are they new Properties?

marktrayer replied:
Swagger keywords.

marktrayer replied:
Swagger constructs

marktrayer committed:
Addressing comments from Mitch

marktrayer committed:
Fix merge conflict
Merge branch 'Bugzilla-2666' of https://github.com/openconnectivityfoundation/bridging into Bugzilla-2666

marktrayer committed:
Update oic.r.vodlist.swagger.json
